config/pipeline.yaml

```diff
@@ -908,6 +908,162 @@ jobs:
       job_timeout: 10
     kcidb_test_suite: kselftest.iommu
 
+  ltp: &ltp
+    template: ltp.jinja2
+    kind: job
+    nfsroot: 'https://storage.kernelci.org/images/rootfs/debian/bookworm-ltp/20240313.0/{debarch}'
+    params: &ltp-params
+      skip_install: "true"
+      tst_cmdfiles: ""
+      job_timeout: '15'
+      skipfile: skipfile-lkft.yaml
+    kcidb_test_suite: ltp
+    rules:
+      - blocklist: &kselftest_defconfig_filter
+        defconfig: ['kselftest']
+
+  ltp-cap-bounds:
+    <<: *ltp
+    params:
+      <<: *ltp-params
+      tst_cmdfiles: "cap_bounds"
+
+  ltp-cpuhotplug:
+    <<: *ltp
+    params:
+      <<: *ltp-params
+      tst_cmdfiles: "cpuhotplug"
+
+  ltp-crypto:
+    <<: *ltp
+    params:
+      <<: *ltp-params
+      tst_cmdfiles: "crypto"
+    rules:
+      - passlist:
+          defconfig: ['crypto']
+
+  ltp-dio:
+    <<: *ltp
+    params:
+      <<: *ltp-params
+      tst_cmdfiles: "dio"
+
+  ltp-fcntl-locktests:
+    <<: *ltp
+    params:
+      <<: *ltp-params
+      tst_cmdfiles: "fcntl-locktests"
+
+  ltp-filecaps:
+    <<: *ltp
+    params:
+      <<: *ltp-params
+      tst_cmdfiles: "filecaps"
+
+  ltp-fsx:
+    <<: *ltp
+    params:
+      <<: *ltp-params
+      tst_cmdfiles: "fsx"
+
+  ltp-hugetlb:
+    <<: *ltp
+    params:
+      <<: *ltp-params
+      tst_cmdfiles: "hugetlb"
+
+  ltp-ima:
+    <<: *ltp
+    params:
+      <<: *ltp-params
+      tst_cmdfiles: "ima"
+    rules:
+      - passlist:
+          defconfig: ['ima']
+
+  ltp-io:
+    <<: *ltp
+    params:
+      <<: *ltp-params
+      tst_cmdfiles: "io"
+
+  ltp-ipc:
+    <<: *ltp
+    params:
+      <<: *ltp-params
+      tst_cmdfiles: "ipc"
+
+  ltp-mm:
+    <<: *ltp
+    params:
+      <<: *ltp-params
+      tst_cmdfiles: "mm"
+      job_timeout: '30'
+
+  ltp-pty:
+    <<: *ltp
+    params:
+      <<: *ltp-params
+      tst_cmdfiles: "pty"
+      job_timeout: '25'
+      priority: 0
+
+  ltp-sched:
+    <<: *ltp
+    params:
+      <<: *ltp-params
+      tst_cmdfiles: "sched"
+
+  ltp-smoketest:
+    <<: *ltp
+    params:
+      <<: *ltp-params
+      tst_cmdfiles: "smoketest"
+
+  ltp-syscalls:
+    <<: *ltp
+    params:
+      <<: *ltp-params
+      tst_cmdfiles: "syscalls"
+      job_timeout: '120'
+      priority: 0
+
+  ltp-timers:
+    <<: *ltp
+    params:
+      <<: *ltp-params
+      grp_test: "TMR"
+      job_timeout: '30'
+      priority: 0
+
+  ltp-timers_qemu:
+    <<: *ltp
+    base_name: ltp-timers
+    params:
+      <<: *ltp-params
+      grp_test: "TMR"
+      job_timeout: '30'
+      priority: 0
+    rules:
```

Comment:
@a-wai does the rule have the same combination, passlist, and blocklist config options as filters had in legacy?

Reply:
@a-wai I'm translating it. How should I specify combination of defconfig+fragment there. If I specify a fragment for one architecture, other arch nodes wouldn't be generated. Any idea? Otherwise I can create different job variants for different architectures.
```
    rules:
      arch:
        - arm
        - arm64
        - i386
        - x86
      defconfig:
        - 'multi_v5_defconfig'
        - 'multi_v7_defconfig'
        - 'defconfig'
        - 'defconfig+arm64-chromebook'
        - 'i386_defconfig'
        - 'x86_64_defconfig'
        - 'x86_64_defconfig+x86-board'
      fragments:
        - '!kselftest'
      lab:
        - 'lab-baylibre'
        - 'lab-broonie'
        - 'lab-collabora'
        - 'lab-collabora-staging'
```

Reply:
Keep in mind that test jobs are scheduled for specific kbuild nodes, so you can add those rules in the corresponding `scheduler` entries:
* for arm64 kbuilds, you'll allow `defconfig` (unless said kbuild node already limits builds to this one)
* for arm builds, you'll allow `defconfig` and the "multi" ones
* etc

Same for the `lab` entries: jobs are scheduled per-labs, so this is not needed here.

The only rule you can specify for the job itself is the `fragments` one.

Reply:
After creating so many complex filtered job, it is only ran on x86 qemu in legacy. I'll remove all other architectures from the job and keep only for qemu_x86.